update sonarqube docker image

muhlba91 opened the pull request:
Docker image update to 7.3 but based on alpine.
The reason is that SonarQube officially stopped distributing Docker images due to the different versions. Then, the latest official repository version of the alpine version has been forked and continued with the latest SonarQube versions starting with 7.3.

rnveach commented:
TeamCity failure is `Unable to load the mojo 'test' (or one of its required components) from the plugin 'org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M1'`.
I will try restarting.

rnveach commented:
Re-run produced the same result.

muhlba91 commented:
It reminds me about the issue I had in the other branch with this build, which was caused by not specifying an explicit surefire plugin version since version 3.x doesn‘t seem to work with the chosen maven version or due to some other reason.
I fixed it by including the surefire plugin in the latest version before 3.x as you can see at line 230 in the new pom.xml from the other PR.

romani commented:
>FROM muhlba91/sonarqube:7.3

Please give us a link to source of this image, we need to copy it to organization source repository, we can not depend of some private resources.

muhlba91 commented:
Sure. Here it is: https://github.com/muhlba91/sonarqube-docker
And as a reference, the latest 7.1-alpine from SonarSource: https://github.com/SonarSource/docker-sonarqube/tree/master/7.1-alpine

muhlba91 commented:
@romani any update on this?
btw, would it be possible for Wercker to specify a Dockerfile? Maybe we could integrate it like this easier and you wouldn't depend on any other maintainer for your containers?

romani commented:
Sorry for delay , your update is in my top Todo list, but I need to allocate some laptop time to deploy new image to our dicker repo

romani commented:
@muhlba91 , sorry for delay.

Please explain me one more time why you think we should have  custom docker file , and can not simply be based on what sonarqube  provide?
https://hub.docker.com/_/sonarqube/
https://github.com/SonarSource/docker-sonarqube
https://github.com/SonarSource/docker-sonarqube/blob/master/7.4-community-alpine/Dockerfile

muhlba91 commented:
Hi,

oh, that must be some newer build. The last time I checked there were no official builds (see https://github.com/SonarSource/docker-sonarqube/issues/196).
Then we‘ll surely should use the official build now!

romani commented:
Please send PR

muhlba91 commented:
I updated this PR. Since there's now an official SonarQube Docker image, I updated it to the earliest version available of 7.2+, which is 7.4-community.

romani commented:
@muhlba91 , did you test image build on this change ?
I have problem to build:
```
$ docker build -t sonarqube-maven-git .
Sending build context to Docker daemon  244.7kB
Step 1/10 : FROM sonarqube:7.4-community
 ---> 471eeabfec42
Step 2/10 : ARG MAVEN_VERSION=3.3.9
 ---> Using cache
 ---> 055e8eaedd1f
Step 3/10 : ARG USER_HOME_DIR="/root"
 ---> Using cache
 ---> 8b3b45e5dcab
Step 4/10 : RUN mkdir -p /usr/share/maven /usr/share/maven/ref   && curl -fsSL http://apache.osuosl.org/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-$MAVEN_VERSION-bin.tar.gz     | tar -xzC /usr/share/maven --strip-components=1   && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
 ---> Running in 8d1980e7c598
mkdir: cannot create directory ‘/usr/share/maven’: Permission denied
mkdir: cannot create directory ‘/usr/share/maven’: Permission denied
The command '/bin/sh -c mkdir -p /usr/share/maven /usr/share/maven/ref   && curl -fsSL http://apache.osuosl.org/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-$MAVEN_VERSION-bin.tar.gz     | tar -xzC /usr/share/maven --strip-components=1   && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn' returned a non-zero code: 1
✘-1 
```

romani commented:
issue is fixed - https://github.com/checkstyle/sonar-checkstyle/commit/b7469ff8b2d049ad7f116efda6dfe1ccb5120907
